kapitan/resources.py

```diff
@@ -250,3 +252,13 @@ def inventory_reclass(inventory_path):
             raise InventoryError(e.message)
 
     return cached.inv
+
+
+def reveal_or_compile_refs(inv):
```

Comment:
I like the consolidation of this behaviour to just one function instead of the previous duplication in code.
But if I understand this correctly it will always by default reveal when compiling ? If that's the case, that'd be a breaking change and it might break some user cases as well. 

I might be misinterpreting it, but isn't issue #341 already fixed in one of our examples?  
https://github.com/deepmind/kapitan/blob/master/examples/kubernetes/components/mysql/secret.jsonnet#L9
password declared here: https://github.com/deepmind/kapitan/blob/master/examples/kubernetes/inventory/classes/component/mysql.yml#L43 
I think you just need the same thing, but without the `|sha256` in the secret generation ? 
We already have several secret values in k8s secrets this way.

Reply:
It should not be a breaking change since it will still reveal or compile the refs in the inventory based on the arguments given to the command. The reveal argument is passed to the Revealer when it is instantiated now.

However, I did realize there was one issue with the implementation this morning. I need to filter the reveal down to specific targets if they are specified. Otherwise, it will always try to reveal the entire inventory, which may not always be possible (e.g., I only have access to the decryption secrets for a subset of the targets). I'll push up the necessary changes soon.

The examples unfortunately do not resolve the issue. It works great when you only want to set single key/value pairs in a Secret; however, if I want to build an entire configuration file from the inventory data and only encrypt one value, then it is not possible to do so. In the current state, you either need to encrypt the entire file or nothing at all.

Reply:
@uberspot I've updated the logic to include only revealing or compiling specified targets rather than always doing it for all targets. I've included some additional testing for the logic as well. 

Additionally, if you take a look at https://github.com/deepmind/kapitan/pull/363/files#diff-7626220eb9df789da70769fc68e3580aR313, you can see that the reveal argument is indeed passed to the Reveal object at instantiation. Then used here https://github.com/deepmind/kapitan/pull/363/files#diff-f524753f276883cad143588359f4fde7R276-R279 to flex between revealing and compiling.